src/include/btree.i

```diff
@@ -981,8 +981,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * been written in the checkpoint, leaving the checkpoint inconsistent.
 	 */
 	if (btree->checkpointing &&
-	    (__wt_page_is_modified(page) ||
-	    F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) {
+	    (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MASK))) {
```

Comment:
> Another thought I had here is to have __rec_child_modified update the parent page in memory and clear WT_PM_REC_MASK on child pages where possible. That's not simple for splits, though.

That interesting, I agree -- I'm not so worried about splits, if we can do it for "normal" pages we'll be back to where we were.